### PR TITLE
add create_customer in routes

### PR DESCRIPTION
**Pull request by Andision**
#11

### CODE REVIEW

**Comment by Andision** [comment]
```
Name                               Stmts   Miss  Cover   Missing
----------------------------------------------------------------
service/__init__.py                   17      2    88%   31-32
service/common/cli_commands.py         7      0   100%
service/common/error_handlers.py      38     19    50%   32, 38-40, 51-53, 62-64, 77-79, 92-94, 107-109
service/common/log_handlers.py        11      1    91%   35
service/common/status.py              45      0   100%
service/config.py                      5      0   100%
service/models.py                     65      0   100%
service/routes.py                     19      0   100%
----------------------------------------------------------------
TOTAL                                207     22    89%
```

**Review by elien2016 — COMMENTED** [review]
For create_customer() in routes.py, the response returned must include a location field in the header, e.g. "{"Location": location_url}", other things there look good. Maybe consider append some messages to the logger though, not required but would be nice:)

File: tests/test_routes.py [hunk 1]
```diff
@@ -43,3 +47,107 @@ def test_index(self):
...
+        self.assertEqual(new_json["email"], fake_customer.email)
+        self.assertEqual(new_json["address"], fake_customer.address)
+
+    def test_create_customer_not_json(self):
+        """ Create a new Customer with non-JSON data """
+        # Arrange
+        fake_customer = CustomerFactory()
+
+        # Convert the data to a string instead of JSON
+        data = str({
```
Comment (elien2016) [inline tests/test_routes.py:101]: please call Customer.serialize() instead of str(), i.e. data = fake_customer.serialize()

**Review by elien2016 — COMMENTED** [review]
nice job testing for missing fields, consider adding tests for no data (empty json), no content type (no data at all), wrong content type, wrong field types

**Comment by Andision** [comment]
all tested

```
vscode@nyu:/app$ make test
Running tests...
green -vvv --processes=1 --run-coverage --termcolor --minimum-coverage=95
Green 3.4.3, Coverage 7.1.0, Python 3.11.5

tests.test_cli_commands
  TestFlaskCLI
.   It should call the db-create command
tests.test_models
  TestCustomer
.   It should Create a customer and add it to the database
.   It should Create a customer and assert that it exists
.   It should Delete a Customer
.   It should de-serialize a Customer
.   It should not deserialize bad data
.   It should not deserialize a Customer with missing data
.   It should Find a Customer by Name
.   It should Find a Customer by ID
.   It should List all Customers in the database
.   It should Read a Customer
.   It should serialize a Customer
.   It should Update a Customer
.   It should not Update a Customer with no id
tests.test_routes
  TestCustomerServer
.   Create a new Customer
.   Create a new Customer with missing fields
.   Create a new Customer with no JSON data
.   Create a new Customer with no content_type
.   Create a new Customer with non-JSON data
.   Create a new Customer with wrong field
.   test_get_customer
.   It should call the home page

Name                               Stmts   Miss  Cover   Missing
----------------------------------------------------------------
service/__init__.py                   17      2    88%   31-32
service/common/cli_commands.py         7      0   100%
service/common/error_handlers.py      38     13    66%   32, 38-40, 62-64, 77-79, 107-109
service/common/log_handlers.py        11      1    91%   35
service/common/status.py              45      0   100%
service/config.py                      5      0   100%
service/models.py                     65      0   100%
service/routes.py                     36      0   100%
----------------------------------------------------------------
TOTAL                                224     16    93%

Ran 22 tests in 0.757s using 1 process

Coverage of 93% is below minimum level of 95%
FAILED (passes=22)
Coverage of 93% is below minimum level of 95%
```

**Comment by elien2016** [comment]
great job
